api/src/controllers/assets.ts

```diff
@@ -23,6 +24,50 @@ const env = useEnv();
 
 router.use(useCollection('directus_files'));
 
+router.post(
+	'/folder/:pk',
+	asyncHandler(async (req, res) => {
+		const service = new AssetsService({
+			accountability: req.accountability,
+			schema: req.schema,
+		});
+
+		const { archive, complete } = await service.zipFolder(req.params['pk']!);
+
+		res.setHeader('Content-Type', 'application/zip');
+		res.setHeader('Content-Disposition', `attachment; filename="folder-${new Date().toISOString()}.zip"`);
```

Comment:
Perhaps we could inject the folder name into the filename so that the user can more easily distinguish it?

Reply:
I assume you're referring to using the root folder name for the ZIP file. I would say this depends on how we structure the zip.

Currently the root folder appears as a top-level directory inside the ZIP, so naming the ZIP after it could be confusing. If we instead remove the root folder and place its files/folders directly at the top level of the ZIP, then using the root folder’s name could work

Additionally for files we do not have this option, at least this way they are consistent.

Reply:
My idea was to keep the folder prefix but add the folderName if it exists

Something like
```ts
`attachment; filename="folder${folderName ? `${folderName}-` : '-'}${getDateFormatted()}}.zip"` ?
```

Reply:
Added in 90405b32f3c7ae7b1b29c58ffbfe1f13fc83ca7a